pkg/types/vsphere/validation/machinepool.go

```diff
@@ -25,18 +32,27 @@ func ValidateMachinePool(platform *vsphere.Platform, machinePool *types.MachineP
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("coresPerSocket"), vspherePool.NumCoresPerSocket, "cores per socket must be positive"))
 	}
 
-	defaultCoresPerSocket := int32(4)
-	defaultNumCPUs := int32(4)
+	// Either the number set by the user or a default value
+	var numCPUs int32
 	if vspherePool.NumCPUs > 0 {
-		if vspherePool.NumCoresPerSocket > vspherePool.NumCPUs {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("coresPerSocket"), vspherePool.NumCoresPerSocket, "cores per socket must be less than number of CPUs"))
-		} else if vspherePool.NumCoresPerSocket > 0 && vspherePool.NumCPUs%vspherePool.NumCoresPerSocket != 0 {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("cpus"), vspherePool.NumCPUs, "numCPUs specified should be a multiple of cores per socket"))
-		} else if vspherePool.NumCPUs%defaultCoresPerSocket != 0 {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("cpus"), vspherePool.NumCPUs, "numCPUs specified should be a multiple of cores per socket which is by default 4"))
-		}
-	} else if vspherePool.NumCoresPerSocket > defaultNumCPUs {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("coresPerSocket"), vspherePool.NumCoresPerSocket, "cores per socket must be less than number of CPUs which is by default 4"))
+		numCPUs = vspherePool.NumCPUs
+	} else {
+		numCPUs = defaultNumCPUs
```

Comment:
nit: since you are checking with `vspherePool.NumCPUs` you could drop this else and assign before the if.

same suggestion to the below `vspherePool.NumCoresPerSocket`